docs/getting-started/built-ins.md

```diff
@@ -153,9 +153,75 @@ rshift(1024n, 11n) // 0n
 
 - `exit(status: boolean): void` Call this function will terminate contract execution. If `status` is `true` then the contract succeeds; otherwise, it fails.
 
+## `SmartContract` Functions
+
+The following functions come with the `SmartContract` base class.
+
+### `this.buildStateOutput(amount: bigint)`
+
+Function `this.buildStateOutput` creates an output containing the latest state. It takes an input: the number of satoshis in the output. With this function, you don't need to construct the new state output manually in the contract.
+
+```typescript
+class Counter extends SmartContract {
+  // ...
+
+  @method(SigHash.ANYONECANPAY_SINGLE)
+  public incOnChain() {
+    // ... update state
+      
+    // construct the new state output 
+    const output: ByteString = this.buildStateOutput(this.ctx.utxo.value)
+
+    // ... verify outputs of current tx
+  }
+}
+```
+
+### `this.buildChangeOutput()`
+
+Function `this.buildChangeOutput` creates a P2PKH change output for the method calling transaction, with no parameters required. It will calculate the change amount automatically, and use the default address of the signer as the change address by default, or parse it from `changeAddress` field in `MethodCallOptions`.
```

Comment:
Function `buildChangeOutput` creates a P2PKH change output. It will calculate the change amount automatically, and use the signer's address by default, unless `changeAddress` field is explicitly set in `MethodCallOptions`.

Reply:
Also document `this.changeAmount
`https://github.com/sCrypt-Inc/scryptTS-examples/blob/733e248ca63ca3506e0d0eb48688b3ffc1d9f8ca/src/contracts/recallable.ts#L90